readium/streamer/src/main/java/org/readium/r2/streamer/parser/PublicationParser.kt

```diff
@@ -20,7 +21,18 @@ import org.readium.r2.shared.util.mediatype.MediaType
  */
 public interface PublicationParser {
 
+    /**
+     * Full publication asset.
+     *
+     * @param sourceAsset Asset of the source used to build the publication. It can be a package,
+     * a JSON manifest, a LCP license, etc.
+     * @param mediaType Media type of the "virtual" publication asset, built from the source asset.
+     * For example, if the source asset was a `application/audiobook+json`, the "virtual" asset
+     * media type will be `application/audiobook+zip`.
+     * @param container Container granting access to the resources of the publication.
+     */
     public data class Asset(
+        val sourceAsset: SharedAsset,
```

Comment:
Added the `sourceAsset` because it was needed by the `ReadiumWebPubParser` to differentiate between a packaged and a manifest when resolving hrefs.